azerothcore/azerothcore-wotlk
# fix(Core): Implement retail-like fishing skill-up functionality

**This Pull request was inspired by [this trinitycore pull request](https://github.com/TrinityCore/TrinityCore/pull/17528) but another solution was implemented after figuring out their pull request is wrong.**

## Changes Proposed:
- Fishing now levels up based on a probability to simulate the retail levelling up process!
- It used to take 2 Hours to get 450 fishing on current Azerothcore but now it will take an average of 12 hours which is equivalent to 2604 catches in total!

> At the start, each catch will increase your skill by one. As you become more advanced it will take more and more catches to advance another point. The table below shows the approximate number of successful catches required to increase your fishing skill by a point. The process is random: The number of catches per skill increase is not constant. Use the table as a guide only.

[El's Extreme Angling](https://web.archive.org/web/20100305183419/http://www.elsanglin.com/role_of_skill.html)
[Wowwiki page on fishing](https://wowwiki-archive.fandom.com/wiki/Fishing?oldid=2356392)

| From fishing | to fishing | Attempts per catch ON AVERAGE|
| ------ | ------ | -------------------------------- |
|1	|115	|1
|115	|135	|2
|135	|160	|3
|160	|190	|4
|190	|215	|5
|215	|295	|6
|295	|315	|9
|315	|355	|10
|355	|425	|11
|425	|450	|12

## Issues Addressed:
Not reported



## Tests Performed:
- Compiled and tested on Windows 10



## How to Test the Changes:

- Create an Alliance character
- .go c 79835
- .modify money 1000000
- Learn fishing from Arnold Leland
- Grab a Fishing rod from Catherine Leland
- Cast the Fishing spell from your general spellbook
- Fish from 1-75 and get 1 level per catch
- .set 356 75
- Learn Journeyman Fishing from the trainer
- .set 356 115
- Cast the Fishing spell and see that it takes **ON AVERAGE** 2 catches in order to level up **once**
- .set 356 125
- Learn Expert Fishing
- .set 356 200
- Learn Artisan Fishing
- .set 356 214
- Cast the Fishing spell and see that it takes **ON AVERAGE** 5 catches in order to level up **once**

For a better test result, I suggest fishing for 10 Skill ups, counting the amount of fish caught and then dividing by the average number on the table. Example:
At 300 Fishing, I swung 22 times in order to level up twice which is an 11% chance per catch. 11% catch is an average of 100/9 (100 per cent chance/number of average fish aka steps)

## Known Issues and TODO List:

Nothing left to do

## Target Branch(es):








 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
